internal/metricdata/servicestats.go

```diff
@@ -0,0 +1,109 @@
+// SPDX-FileCopyrightText: 2022-present Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package metricdata
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/omec-project/metricfunc/logger"
+	"github.com/omec-project/metricfunc/pkg/metricinfo"
+)
+
+type nfServiceStats struct {
+	svcStatLock sync.RWMutex
+	svcStats    map[string]map[string]uint64 //Nf IP is key
+}
+
+func HandleServiceEvent(msgType *metricinfo.CoreMsgType, sourceNf metricinfo.NfType) {
+	switch sourceNf {
+	case metricinfo.NfTypeSmf:
+		handleSmfServiceEvent(msgType)
+	case metricinfo.NfTypeAmf:
+		handleAmfServiceEvent(msgType)
+	default:
+		logger.CacheLog.Errorf("unknown msg source [%v] ", sourceNf)
+	}
+}
+
+func handleSmfServiceEvent(msgType *metricinfo.CoreMsgType) {
+
+	metricData.SmfSvcStats.svcStatLock.Lock()
+	defer metricData.SmfSvcStats.svcStatLock.Unlock()
+
+	//Check if bucket already exist
+	if stats, ok := metricData.SmfSvcStats.svcStats[msgType.SourceNfIp]; ok {
+		stats[msgType.MsgType] = stats[msgType.MsgType] + 1
+		return
+	}
+
+	stat := make(map[string]uint64)
+	stat[msgType.MsgType] = 1
+	metricData.SmfSvcStats.svcStats[msgType.SourceNfIp] = stat
```

Comment:
IP address based map seems vulnerable. Since K8s reuses addresses as PODs come up and go. 
Comment applicable at all other places as well.

Reply:
Yes, IP-Address may be reused on pod restarts but it gives more detailed view of what prior smf handled. We can discuss this and can avoid stats per instance level. This shall have stats at pure service level rather than instance level.

Reply:
Use Nf ID instead of SmfIP